### PR TITLE
Support mtp run in full graph mode

### PR DESCRIPTION
**Pull request by anon189Ty**
### What this PR does / why we need it?
Currently, the MTP model still runs in eager in full graph mode. This PR adapts the MTP with the full graph capture and execution. When the graph mode is set to "FULL_DECODE_ONLY", the MTP will run in full-graph to improve the performance.

The change include:
1. Add _mtp_graph_params in acl_graph.py to isolate the data of main model and the data of mtp.
2. Padding some metadata in mla_v1.py when in fullgraph mode.
3. Fixed the essential data address that will be used in model.forward.
4. Adapted according to the aclgraph capture framwork:
    1). Rebuild mtp model with ACLGraphWrapper.
    2). Add common attn metadata when start capture in mtp dummy_run.
    3). Add common attn metadata update in mtp.
    4). Addapted data update when num_speculative_tokens > 1.

### CODE REVIEW

File: vllm_ascend/spec_decode/mtp_proposer.py [hunk 1]
```diff
@@ -321,6 +411,17 @@ def _prepare_inputs(
...
+
+        batch_size = num_rejected_tokens.shape[0]
+        self.query_start_loc[:batch_size + 1].copy_(cu_num_tokens[:batch_size +
+                                                                  1])
+        self.query_start_loc_cpu[:batch_size + 1].copy_(
+            self.query_start_loc[:batch_size + 1], non_blocking=True)
+        target_positions_len = target_positions.shape[0]
+        self.positions[:target_positions_len].copy_(target_positions)
+        target_slot_mapping_len = target_slot_mapping.shape[0]
+        self.slot_mapping[:target_slot_mapping_len].copy_(target_slot_mapping)
```
Comment (JC-ut0) [inline vllm_ascend/spec_decode/mtp_proposer.py]: I am wondering the performance impact of those `copy` operations

File: vllm_ascend/spec_decode/mtp_proposer.py [hunk 2]
```diff
@@ -71,6 +75,24 @@ def __init__(
         self.use_sparse = hasattr(vllm_config.model_config.hf_config,
                                   "index_topk")
 
+        self.query_start_loc = torch.zeros(
```
Comment (whx-sjtu) [inline vllm_ascend/spec_decode/mtp_proposer.py]: Maybe these pinned tensors can also reuse corresponding tensors in model runner by reference to `self.runner`?

**Comment by JC-ut0** [comment]
Plead add UT tests for FULL_DECODE_ONLY aclgraph

**Comment by wangxiyuan** [comment]
has this change be merged into main? which commit

**Comment by yiz-liu** [comment]
This PR isn’t needed anymore since it’s already implemented on the main branch, and no new features should be added to the v0.11.0-dev release.
